server/src/main/java/org/elasticsearch/index/mapper/AbstractScriptFieldType.java

```diff
@@ -204,12 +204,13 @@ protected final LeafFactory leafFactory(SearchExecutionContext context) {
     }
 
     @Override
-    public void validateMatchedRoutingPath() {
+    public void validateMatchedRoutingPath(final String routingPath) {
         throw new IllegalArgumentException(
-            "All fields that match routing_path must be keywords with [time_series_dimension: true] "
+            "All fields that match routing_path "
```

Comment:
I think I need to update these error message sto include also "time_series_dimensions" for flattened fields.